### PR TITLE
[FSDP][state_dict] Add world_size 1 unittest

### PR DESCRIPTION
**Pull request by fegin**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #112669

As title

Differential Revision: [D50754433](https://our.internmc.facebook.com/intern/diff/D50754433/)

### CODE REVIEW

**Review by wz337 — APPROVED** [review]
LGTM!

File: test/distributed/fsdp/test_fsdp_state_dict.py [hunk 1]
```diff
@@ -1212,6 +1213,25 @@ def test_sharded_load_multi_backend_pg(self):
...
+        my_pg = None
+        for i in range(self.world_size):
+            pg = dist.new_group(ranks=[i])
+            if i == self.rank:
+                my_pg = pg
+
+        model = TransformerWithSharedParams.init(
+            my_pg,
+            FSDPInitMode.RECURSIVE,
+            CUDAInitMode.CUDA_BEFORE,
```
[the remark this answers is not in the document]
Comment (fduwjj) [reply test/distributed/fsdp/test_fsdp_state_dict.py:1227]: it is explained here: https://github.com/pytorch/pytorch/blob/6e94fc465e8ebb2424bc8207a025e9085bcc0fe7/torch/testing/_internal/common_fsdp.py#L424

File: test/distributed/fsdp/test_fsdp_state_dict.py [hunk 2]
```diff
@@ -132,6 +132,23 @@ def forward(self, x):
         return self.outer(i + j)
 
 
+class TestDummyModel(torch.nn.Module):
```
Comment (fduwjj) [inline test/distributed/fsdp/test_fsdp_state_dict.py:135]: not necessary in this PR, but can we move it to torch/testing/_internal/common_fsdp.py?

**Review by fduwjj — APPROVED** [review]

**Pull request closed without merging**